superset/views/geocoding.py

```diff
@@ -230,200 +223,174 @@ def geocode(self) -> Response:
+        if city_column:
+            columns.append(city_column)
+        if country_column:
+            columns.append(country_column)
+        if building_number_column:
+            columns.append(building_number_column)
+        columns.append(request_data.get("latitudeColumnName", "lat"))
+        columns.append(request_data.get("longitudeColumnName", "lon"))
+        return columns
+
+    def _get_from_clause(self, table: SqlaTable) -> str:
+        """
+        Get the from clause for the SQL statement
+        :param table: The SqlaTable
+        :return: The from clause as a string
+        """
         quote = table.database.get_sqla_engine().dialect.identifier_preparer.quote
         if table.schema:
             full_table_name = quote(table.schema) + "." + quote(table.table_name)
         else:
             full_table_name = quote(table.table_name)
         return text(full_table_name)
 
-    def _check_and_create_columns(self, request_data):
-        lat_column = request_data.get("latitudeColumnName", "lat")
-        lon_column = request_data.get("longitudeColumnName", "lon")
-        if_exists = request.json.get("overwriteIfExists", "fail")
-        if "fail" in if_exists:
-            override_if_exist = False
-        else:
-            override_if_exist = True
-        table_dto = request_data.get("datasource", models.TableDto())
-        table_id = table_dto.get("id", "")
-        table = self._get_table_by_id(table_id)
-        lat_exists = self._does_column_name_exist(table, lat_column)
-        lon_exists = self._does_column_name_exist(table, lon_column)
-        column_message_suffix = ""
-        if override_if_exist:
-            if lat_exists:
-                if lon_exists:
-                    pass
-                else:
-                    self._add_lat_lon_columns(table, lon_column=lon_column)
-                    column_message_suffix = f"successfully created column {lon_column}"
-            else:
-                if lon_exists:
-                    self._add_lat_lon_columns(table, lat_column=lat_column)
-                    column_message_suffix = f"successfully created column {lat_column}"
-                else:
-                    self._add_lat_lon_columns(table, lat_column, lon_column)
-                    column_message_suffix = (
-                        f"successfully created columns: {lat_column} and {lon_column}"
-                    )
-        else:
-            if lat_exists:
-                raise ValueError(
-                    "Column name {0} for latitude is already in use".format(lat_column)
-                )
-            if lon_exists:
-                raise ValueError(
-                    "Column name {0} for longitude is already in use".format(lon_column)
-                )
-            self._add_lat_lon_columns(table, lat_column, lon_column)
-            column_message_suffix = (
-                f"successfully created columns: {lat_column} and {lon_column}"
+    def _create_columns(
```

Comment:
docstring still says check and create but name implies it only creates without any checking?